app/src/lib/stores/repositories-store.ts

```diff
@@ -334,4 +335,62 @@ export class RepositoriesStore extends BaseStore {
       repository.missing
     )
   }
+
+  public async updateLastPruneDate(repository: Repository): Promise<void> {
+    const repoID = repository.id
+    if (!repoID) {
+      return fatalError(
+        '`updateLastPruneDate` can only update the last prune date for a repository which has been added to the database.'
+      )
+    }
+
+    const githubRepo = repository.gitHubRepository
+    if (githubRepo === null) {
+      return fatalError(
+        `'updateLastPruneDate' can only update GitHub repositories`
```

Comment:
I might be reading this wrong but this looks like it's saying that we'll only prune branches for GitHub repositories. Why is that?

Reply:
We initially decided to support only GH repos because we want to be sure we aren't deleting branches we shouldn't be deleting. Since we rely on the default branch to determine what's been merged, we have to ensure we have the correct default branch.